db/class.go

```diff
@@ -224,8 +228,31 @@ func (d *DB) GetClass(c echo.Context) error {
 		return c.String(http.StatusNotFound, "given user not in class")
 	}
 
+	if WithPrograms := c.QueryParam("programs"); WithPrograms != "" {
```

Comment:
If you can think of a better way of gating this functionality, that would be great, but if not this is okay.